diff --git cpp/src/arrow/io/mman.h cpp/src/arrow/io/mman.h
--- cpp/src/arrow/io/mman.h
+++ cpp/src/arrow/io/mman.h
@@ -3,0 +3,3 @@
+#ifdef _WIN32
+typedef long long off_t;
+#endif
diff --git cpp/src/plasma/protocol.cc cpp/src/plasma/protocol.cc
--- cpp/src/plasma/protocol.cc
+++ cpp/src/plasma/protocol.cc
@@ -760,1 +760,5 @@
+#ifdef _WIN32
+  *address = _strdup(message->address()->c_str());
+#else
   *address = strdup(message->address()->c_str());
+#endif
-- 
