diff --git bazel/glog.bzl bazel/glog.bzl
--- bazel/glog.bzl
+++ bazel/glog.bzl
@@ -19,3 +19,2 @@ def glog_library(namespace = "google", with_gflags = 1, **kwargs):
         "-DHAVE_STRING_H",
-        "-DHAVE_UNWIND_H",
     ] + (["-DHAVE_LIB_GFLAGS"] if with_gflags else [])
@@ -38,5 +37,7 @@ def glog_library(namespace = "google", with_gflags = 1, **kwargs):
         # Enable dumping stacktrace upon sigaction.
         "-DHAVE_SIGACTION",
+        "-DHAVE_EXECINFO_H",
+        "-DHAVE_SYMBOLIZE",
         # For logging.cc.
         "-DHAVE_PREAD",
         "-DHAVE___ATTRIBUTE__",
-- 
