diff --git src/core/lib/promise/pipe.h b/core/lib/promise/pipe.h
--- src/core/lib/promise/pipe.h
+++ src/core/lib/promise/pipe.h
@@ -63,9 +63,9 @@
 template <typename T>
 class NextResult final {
  public:
-  NextResult() : center_(nullptr) {}
+  NextResult() : center_(nullptr), cancelled_(true) {}
   explicit NextResult(RefCountedPtr<pipe_detail::Center<T>> center)
-      : center_(std::move(center)) {
+      : center_(std::move(center)), cancelled_(false) {
     CHECK(center_ != nullptr);
   }
   explicit NextResult(bool cancelled)
