diff -u src/core/lib/experiments/experiments.cc
--- src/core/lib/experiments/experiments.cc
+++ src/core/lib/experiments/experiments.cc
@@ -363,7 +363,7 @@
      additional_constraints_event_engine_callback_cq,
      required_experiments_event_engine_callback_cq, 1, true, true},
     {"event_engine_client", description_event_engine_client,
-     additional_constraints_event_engine_client, nullptr, 0, true, true},
+     additional_constraints_event_engine_client, nullptr, 0, false, true},
     {"event_engine_dns", description_event_engine_dns,
      additional_constraints_event_engine_dns, nullptr, 0, true, false},
     {"event_engine_dns_non_client_channel",
@@ -371,7 +371,7 @@
      additional_constraints_event_engine_dns_non_client_channel, nullptr, 0,
      false, false},
     {"event_engine_listener", description_event_engine_listener,
-     additional_constraints_event_engine_listener, nullptr, 0, true, true},
+     additional_constraints_event_engine_listener, nullptr, 0, false, true},
     {"free_large_allocator", description_free_large_allocator,
      additional_constraints_free_large_allocator, nullptr, 0, false, true},
     {"local_connector_secure", description_local_connector_secure,
