diff --git third_party/py/python_configure.bzl third_party/py/python_configure.bzl
index 2cb84077e5..e733b9204e 100644
--- third_party/py/python_configure.bzl
+++ third_party/py/python_configure.bzl
@@ -190,7 +190,7 @@ def _get_python_lib(repository_ctx, python_bin, lib_path_key):
         "for path in all_paths:\n" + "  if os.path.isdir(path):\n" +
         "    paths.append(path)\n" + "if len(paths) >=1:\n" +
         "  print(paths[0])\n" + "END")
-    cmd = '%s - %s' % (python_bin, print_lib)
+    cmd = '"%s" - %s' % (python_bin, print_lib)
     result = repository_ctx.execute([_get_bash_bin(repository_ctx), "-c", cmd])
     return result.stdout.strip('\n')
 
-- 
