diff --git third_party/py/python_configure.bzl third_party/py/python_configure.bzl
--- third_party/py/python_configure.bzl
+++ third_party/py/python_configure.bzl
@@ -345,7 +345,7 @@ def _python_autoconf_impl(repository_ctx):
         repository_ctx,
         "_python3",
         _PYTHON3_BIN_PATH,
-        "python3",
+        "python3" if not _is_windows(repository_ctx) else "python.exe",
         _PYTHON3_LIB_PATH,
         False
     )
