diff --git a/async.c b/async.c
index 0018e37..e632705 100644
--- a/async.c
+++ b/async.c
@@ -33,7 +33,9 @@
 #include "alloc.h"
 #include <stdlib.h>
 #include <string.h>
+#if !defined(WIN32) || defined(__MINGW32__)
 #include <strings.h>
+#endif
 #include <assert.h>
 #include <ctype.h>
 #include <errno.h>
diff --git a/fmacros.h b/fmacros.h
index 3227faa..a84e5f9 100644
--- a/fmacros.h
+++ b/fmacros.h
@@ -9,4 +9,8 @@
 #define _DARWIN_C_SOURCE
 #endif
 
+#if defined(_MSC_VER) && !defined(__clang__) && !defined(strdup)
+#define strdup _strdup
+#endif
+
 #endif
diff --git a/sds.h b/sds.h
index 13be75a..40ba5b0 100644
--- a/sds.h
+++ b/sds.h
@@ -39,6 +39,12 @@
 #include <stdarg.h>
 #include <stdint.h>
 
+#if defined(_MSC_VER) && !defined(__clang__)
+#pragma pack(push, 1)
+#pragma push_macro("__attribute__")
+#define __attribute__(A)
+#endif
+
 typedef char *sds;
 
 /* Note: sdshdr5 is never used, we just access the flags byte directly.
@@ -72,6 +78,11 @@ struct __attribute__ ((__packed__)) sdshdr64 {
     char buf[];
 };
 
+#if defined(_MSC_VER) && !defined(__clang__)
+#pragma pop_macro("__attribute__")
+#pragma pack(pop)
+#endif
+
 #define SDS_TYPE_5  0
 #define SDS_TYPE_8  1
 #define SDS_TYPE_16 2
