diff --git deps/hiredis/test.c deps/hiredis/test.c
--- deps/hiredis/test.c
+++ deps/hiredis/test.c
@@ -94,1 +94,1 @@
-static redisContext *connect(struct config config) {
+static redisContext *connect_(struct config config) {
@@ -251,1 +251,1 @@
-    c = connect(config);
+    c = connect_(config);
@@ -390,1 +390,1 @@
-    c = connect(config);
+    c = connect_(config);
@@ -471,1 +471,1 @@
-    c = connect(config);
+    c = connect_(config);
@@ -483,1 +483,1 @@
-    c = connect(config);
+    c = connect_(config);
@@ -517,1 +517,1 @@
-    c = connect(config);
+    c = connect_(config);
@@ -552,1 +552,1 @@
-    c = connect(config);
+    c = connect_(config);
@@ -586,1 +586,1 @@
-    redisContext *c = connect(config);
+    redisContext *c = connect_(config);
-- 
