diff --git a/opencensus/stats/internal/delta_producer.cc b/opencensus/stats/internal/delta_producer.cc
index c61b4d9..cae8883 100644
--- a/opencensus/stats/internal/delta_producer.cc
+++ b/opencensus/stats/internal/delta_producer.cc
@@ -115,7 +115,10 @@ void DeltaProducer::Flush() {
 }
 
 DeltaProducer::DeltaProducer()
-    : harvester_thread_(&DeltaProducer::RunHarvesterLoop, this) {}
+    : harvester_thread_(&DeltaProducer::RunHarvesterLoop, this) {
+  absl::MutexLock l(&mu_);
+  thread_started_ = true;
+}
 
 void DeltaProducer::SwapDeltas() {
   ABSL_ASSERT(last_delta_.delta().empty() && "Last delta was not consumed.");
@@ -131,11 +134,37 @@ void DeltaProducer::RunHarvesterLoop() {
   absl::Time next_harvest_time = absl::Now() + harvest_interval_;
   while (true) {
     const absl::Time now = absl::Now();
-    absl::SleepFor(next_harvest_time - now);
+    absl::SleepFor(absl::Seconds(0.1));
     // Account for the possibility that the last harvest took longer than
     // harvest_interval_ and we are already past next_harvest_time.
-    next_harvest_time = std::max(next_harvest_time, now) + harvest_interval_;
-    Flush();
+    if (absl::Now() > next_harvest_time) {
+      next_harvest_time = std::max(next_harvest_time, now) + harvest_interval_;
+      Flush();
+    }
+    {
+      absl::MutexLock l(&mu_);
+      if (!thread_started_) {
+        break;
+      }
+    }
+  }
+}
+
+void DeltaProducer::SetHarvestInterval(const absl::Duration interval) {
+  harvest_interval_ = interval;
+}
+
+void DeltaProducer::Shutdown() {
+  {
+    absl::MutexLock l(&mu_);
+    if (!thread_started_) {
+      return;
+    }
+    thread_started_ = false;
+  }
+  // Join loop thread when shutdown.
+  if (harvester_thread_.joinable()) {
+    harvester_thread_.join();
   }
 }
 
diff --git a/opencensus/stats/internal/delta_producer.h b/opencensus/stats/internal/delta_producer.h
index bec5605..453b4ef 100644
--- a/opencensus/stats/internal/delta_producer.h
+++ b/opencensus/stats/internal/delta_producer.h
@@ -71,6 +71,8 @@ class DeltaProducer final {
   // Returns a pointer to the singleton DeltaProducer.
   static DeltaProducer* Get();
 
+  void Shutdown();
+
   // Adds a new Measure.
   void AddMeasure();
 
@@ -84,6 +86,8 @@ class DeltaProducer final {
   // Flushes the active delta and blocks until it is harvested.
   void Flush() ABSL_LOCKS_EXCLUDED(delta_mu_, harvester_mu_);
 
+  void SetHarvestInterval(const absl::Duration interval);
+
  private:
   DeltaProducer();
 
@@ -100,7 +104,7 @@ class DeltaProducer final {
   // every harvest_interval_.
   void RunHarvesterLoop();
 
-  const absl::Duration harvest_interval_ = absl::Seconds(5);
+  absl::Duration harvest_interval_ = absl::Seconds(5);
 
   // Guards the active delta and its configuration. Anything that changes the
   // delta configuration (e.g. adding a measure or BucketBoundaries) must
@@ -122,6 +126,9 @@ class DeltaProducer final {
   // thread when calling a flush during harvesting.
   Delta last_delta_ ABSL_GUARDED_BY(harvester_mu_);
   std::thread harvester_thread_ ABSL_GUARDED_BY(harvester_mu_);
+
+  mutable absl::Mutex mu_;
+  bool thread_started_ GUARDED_BY(mu_) = false;
 };
 
 }  // namespace stats
diff --git a/opencensus/stats/internal/stats_exporter.cc b/opencensus/stats/internal/stats_exporter.cc
index 7de96d6..2aef19d 100644
--- a/opencensus/stats/internal/stats_exporter.cc
+++ b/opencensus/stats/internal/stats_exporter.cc
@@ -94,26 +94,58 @@ void StatsExporterImpl::ClearHandlersForTesting() {
   handlers_.clear();
 }
 
+void StatsExporterImpl::Shutdown() {
+  {
+    absl::MutexLock l(&mu_);
+    if (!thread_started_) {
+      return;
+    }
+    thread_started_ = false;
+  }
+  // Join loop thread when shutdown.
+  if (t_.joinable()) {
+    t_.join();
+  }
+}
+
 void StatsExporterImpl::StartExportThread() ABSL_EXCLUSIVE_LOCKS_REQUIRED(mu_) {
-  t_ = std::thread(&StatsExporterImpl::RunWorkerLoop, this);
   thread_started_ = true;
+  t_ = std::thread(&StatsExporterImpl::RunWorkerLoop, this);
 }
 
 void StatsExporterImpl::RunWorkerLoop() {
   absl::Time next_export_time = GetNextExportTime();
   while (true) {
     // SleepFor() returns immediately when given a negative duration.
-    absl::SleepFor(next_export_time - absl::Now());
+    absl::SleepFor(absl::Seconds(0.1));
     // In case the last export took longer than the export interval, we
     // calculate the next time from now.
-    next_export_time = GetNextExportTime();
-    Export();
+    if (absl::Now() > next_export_time) {
+      next_export_time = GetNextExportTime();
+      Export();
+    }
+    {
+      absl::MutexLock l(&mu_);
+      if (!thread_started_) {
+        break;
+      }
+    }
   }
 }
 
 // StatsExporter
 // -------------
 
+void StatsExporter::Shutdown() {
+  StatsExporterImpl::Get()->Shutdown();
+  StatsExporterImpl::Get()->ClearHandlersForTesting();
+}
+
+void StatsExporter::ExportNow() {
+  DeltaProducer::Get()->Flush();
+  StatsExporterImpl::Get()->Export();
+}
+
 // static
 void StatsExporter::SetInterval(absl::Duration interval) {
   StatsExporterImpl::Get()->SetInterval(interval);
diff --git a/opencensus/stats/internal/stats_exporter_impl.h b/opencensus/stats/internal/stats_exporter_impl.h
index abbd13e..823471e 100644
--- a/opencensus/stats/internal/stats_exporter_impl.h
+++ b/opencensus/stats/internal/stats_exporter_impl.h
@@ -34,6 +34,7 @@ class StatsExporterImpl {
  public:
   static StatsExporterImpl* Get();
   void SetInterval(absl::Duration interval);
+  void Shutdown();
   absl::Time GetNextExportTime() const;
   void AddView(const ViewDescriptor& view);
   void RemoveView(absl::string_view name);
diff --git a/opencensus/stats/stats_exporter.h b/opencensus/stats/stats_exporter.h
index 6756858..7aebf21 100644
--- a/opencensus/stats/stats_exporter.h
+++ b/opencensus/stats/stats_exporter.h
@@ -45,6 +45,9 @@ class StatsExporter final {
   // Removes the view with 'name' from the registry, if one is registered.
   static void RemoveView(absl::string_view name);
 
+  static void Shutdown();
+  static void ExportNow();
+
   // StatsExporter::Handler is the interface for push exporters that export
   // recorded data for registered views. The exporter should provide a static
   // Register() method that takes any arguments needed by the exporter (e.g. a
