From 78ac107937ee3b94f0b011f9ef0c12b912a222e7 Mon Sep 17 00:00:00 2001
From: easy <g-easy@users.noreply.github.com>
Date: Wed, 14 Nov 2018 16:49:24 +1100
Subject: [PATCH] Rename net_zlib_zlib to com_github_madler_zlib to match gRPC.

diff --git README.md README.md
index be8e683..1dbc917 100644
--- README.md
+++ README.md
@@ -142,1 +142,1 @@ you introduce the following dependencies, if they do not exist yet, to your proj
-* `load_net_zlib_zlib()` to load `net_zlib_zlib` rules for zlib
+* `load_com_github_madler_zlib()` to load `com_github_madler_zlib` rules for zlib
diff --git pull/BUILD.bazel pull/BUILD.bazel
index 6d90c23..d35f63e 100644
--- pull/BUILD.bazel
+++ pull/BUILD.bazel
@@ -26,1 +26,1 @@ cc_library(
-        "@net_zlib_zlib//:z",
+        "@com_github_madler_zlib//:z",
diff --git bazel/repositories.bzl bazel/repositories.bzl
index 258b73f..91398cf 100644
--- bazel/repositories.bzl
+++ bazel/repositories.bzl
@@ -48,3 +48,4 @@ def prometheus_cpp_repositories():
     maybe(
         http_archive,
-        name = "net_zlib_zlib",
+        # Named "com_github_madler_zlib" to match gRPC's import of zlib.
+        name = "com_github_madler_zlib",
-- 
