--- a/src/google/protobuf/BUILD.bazel
+++ b/src/google/protobuf/BUILD.bazel
@@ -141,1 +141,1 @@
-    exec_tools = ["//:protoc"],
+    tools = ["//:protoc"],

--- a/build_defs/internal_shell.bzl
+++ b/build_defs/internal_shell.bzl
@@ -35,1 +35,1 @@
-        exec_tools = tools,
+        tools = tools,
@@ -80,1 +80,1 @@
-        exec_tools = tools,
+        tools = tools,

--- a/objectivec/BUILD.bazel
+++ b/objectivec/BUILD.bazel
@@ -45,1 +45,1 @@
-    exec_tools = ["//:protoc"],
+    tools = ["//:protoc"],
