diff --git deps/jemalloc/configure deps/jemalloc/configure
--- deps/jemalloc/configure
+++ deps/jemalloc/configure
@@ -7094 +7094 @@
-ARFLAGS='crus'
+ARFLAGS='crs'
diff --git deps/jemalloc/Makefile.in deps/jemalloc/Makefile.in
--- deps/jemalloc/Makefile.in
+++ deps/jemalloc/Makefile.in
@@ -370 +370 @@
-	@if ! `cmp -s $< $@` ; then echo "cp $< $<"; cp $< $@ ; fi
+	@if ! `cmp -s $< $@` ; then                  cp $< $@ ; fi
diff --git deps/Makefile deps/Makefile
--- deps/Makefile
+++ deps/Makefile
@@ -45 +45 @@
-	@printf '%b %b\n' $(MAKECOLOR)MAKE$(ENDCOLOR) $(BINCOLOR)$@$(ENDCOLOR)
+#	@printf '%b %b\n' $(MAKECOLOR)MAKE$(ENDCOLOR) $(BINCOLOR)$@$(ENDCOLOR)
@@ -51 +51 @@
-	@printf '%b %b\n' $(MAKECOLOR)MAKE$(ENDCOLOR) $(BINCOLOR)$@$(ENDCOLOR)
+#	@printf '%b %b\n' $(MAKECOLOR)MAKE$(ENDCOLOR) $(BINCOLOR)$@$(ENDCOLOR)
@@ -61 +61 @@
-LUA_CFLAGS+= -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC='' $(CFLAGS)
+LUA_CFLAGS+= -O2       -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC='' $(CFLAGS)
@@ -67 +67 @@
-ARFLAGS=rcu
+ARFLAGS=rc
@@ -70 +70 @@
-	@printf '%b %b\n' $(MAKECOLOR)MAKE$(ENDCOLOR) $(BINCOLOR)$@$(ENDCOLOR)
+#	@printf '%b %b\n' $(MAKECOLOR)MAKE$(ENDCOLOR) $(BINCOLOR)$@$(ENDCOLOR)
@@ -75 +75 @@
-JEMALLOC_CFLAGS= -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops $(CFLAGS)
+JEMALLOC_CFLAGS= -std=gnu99       -pipe -g3 -O3 -funroll-loops $(CFLAGS)
@@ -79,2 +79,2 @@
-	@printf '%b %b\n' $(MAKECOLOR)MAKE$(ENDCOLOR) $(BINCOLOR)$@$(ENDCOLOR)
+#	@printf '%b %b\n' $(MAKECOLOR)MAKE$(ENDCOLOR) $(BINCOLOR)$@$(ENDCOLOR)
-	cd jemalloc && ./configure --with-version=5.1.0-0-g0 --with-lg-quantum=3 --with-jemalloc-prefix=je_ --enable-cc-silence CFLAGS="$(JEMALLOC_CFLAGS)" LDFLAGS="$(JEMALLOC_LDFLAGS)"
+	cd jemalloc && ./configure --with-version=5.1.0-0-g0 --with-lg-quantum=3 --with-jemalloc-prefix=je_ --silent            CFLAGS="$(JEMALLOC_CFLAGS)" LDFLAGS="$(JEMALLOC_LDFLAGS)"
diff --git src/Makefile src/Makefile
--- src/Makefile
+++ src/Makefile
@@ -176,3 +176,3 @@
-	@echo ""
+#	@echo ""
-	@echo "Hint: It's a good idea to run 'make test' ;)"
+#	@echo "Hint: It's a good idea to run 'make test' ;)"
-	@echo ""
+#	@echo ""
-- 
