diff --git src/Win32_Interop/Win32_QFork.cpp src/Win32_Interop/Win32_QFork.cpp
--- src/Win32_Interop/Win32_QFork.cpp
+++ src/Win32_Interop/Win32_QFork.cpp
@@ -197,1 +197,8 @@
+#else
+const size_t cAllocationGranularity = 1 << 18;    // comparable to the default allocation threshold of Windows
+#ifdef _WIN64
+const int  cMaxBlocks = static_cast<int>((1LL << 40) / cAllocationGranularity);
+#else
+const int  cMaxBlocks = static_cast<int>((1 << 30) / cAllocationGranularity);
+#endif
 #endif
@@ -716,5 +723,7 @@
 #ifdef USE_DLMALLOC
     WSAPROTOCOL_INFO* protocolInfo = (WSAPROTOCOL_INFO*) dlmalloc(sizeof(WSAPROTOCOL_INFO) * g_pQForkControl->globalData.numfds);
 #elif USE_JEMALLOC
     WSAPROTOCOL_INFO* protocolInfo = (WSAPROTOCOL_INFO*) je_malloc(sizeof(WSAPROTOCOL_INFO) * g_pQForkControl->globalData.numfds);
+#else
+    WSAPROTOCOL_INFO* protocolInfo = (WSAPROTOCOL_INFO*) malloc(sizeof(WSAPROTOCOL_INFO) * g_pQForkControl->globalData.numfds);
 #endif
-- 
