diff --git a/src/google/protobuf/compiler/js/js_generator.cc b/src/google/protobuf/compiler/js/js_generator.cc
index 3de61e8..ea83c82
--- a/src/google/protobuf/compiler/js/js_generator.cc
+++ b/src/google/protobuf/compiler/js/js_generator.cc
@@ -488,6 +488,7 @@ string JSByteGetterSuffix(BytesMode bytes_mode) {
       return "U8";
     default:
       assert(false);
+      return NULL;
   }
 }
 
diff --git a/vsprojects/libprotobuf.vcxproj b/vsprojects/libprotobuf.vcxproj
new file
index 0000000..73f9bea
--- /dev/null
+++ b/vsprojects/libprotobuf.vcxproj
@@ -0,0 +1,116 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Label="ProjectConfigurations">
+    <ProjectConfiguration Include="Debug|Win32">
+      <Configuration>Debug</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|Win32">
+      <Configuration>Release</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Debug|x64">
+      <Configuration>Debug</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|x64">
+      <Configuration>Release</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>{07ABD431-1515-4140-932E-E8C150FF7D1A}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <PropertyGroup>
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
+  <ImportGroup Label="PropertySheets">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+    <Import Project="..\..\..\vsprojects\protobuf.props" />
+  </ImportGroup>
+  <ItemGroup>
+    <ClCompile Include="..\src\google\protobuf\arena.cc" />
+    <ClCompile Include="..\src\google\protobuf\arenastring.cc" />
+    <ClCompile Include="..\src\google\protobuf\extension_set.cc" />
+    <ClCompile Include="..\src\google\protobuf\generated_message_util.cc" />
+    <ClCompile Include="..\src\google\protobuf\io\coded_stream.cc" />
+    <ClCompile Include="..\src\google\protobuf\io\zero_copy_stream.cc" />
+    <ClCompile Include="..\src\google\protobuf\io\zero_copy_stream_impl_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\message_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\repeated_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\atomicops_internals_x86_msvc.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\bytestream.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\common.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\int128.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\once.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\status.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\statusor.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\stringpiece.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\stringprintf.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\structurally_valid.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\strutil.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\time.cc" />
+    <ClCompile Include="..\src\google\protobuf\wire_format_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\any.cc" />
+    <ClCompile Include="..\src\google\protobuf\any.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\api.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\importer.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\parser.cc" />
+    <ClCompile Include="..\src\google\protobuf\descriptor.cc" />
+    <ClCompile Include="..\src\google\protobuf\descriptor.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\descriptor_database.cc" />
+    <ClCompile Include="..\src\google\protobuf\duration.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\dynamic_message.cc" />
+    <ClCompile Include="..\src\google\protobuf\empty.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\extension_set_heavy.cc" />
+    <ClCompile Include="..\src\google\protobuf\field_mask.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\generated_message_reflection.cc" />
+    <ClCompile Include="..\src\google\protobuf\io\gzip_stream.cc" />
+    <ClCompile Include="..\src\google\protobuf\io\printer.cc" />
+    <ClCompile Include="..\src\google\protobuf\io\strtod.cc" />
+    <ClCompile Include="..\src\google\protobuf\io\tokenizer.cc" />
+    <ClCompile Include="..\src\google\protobuf\io\zero_copy_stream_impl.cc" />
+    <ClCompile Include="..\src\google\protobuf\map_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\message.cc" />
+    <ClCompile Include="..\src\google\protobuf\reflection_ops.cc" />
+    <ClCompile Include="..\src\google\protobuf\service.cc" />
+    <ClCompile Include="..\src\google\protobuf\source_context.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\struct.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\mathlimits.cc" />
+    <ClCompile Include="..\src\google\protobuf\stubs\substitute.cc" />
+    <ClCompile Include="..\src\google\protobuf\text_format.cc" />
+    <ClCompile Include="..\src\google\protobuf\timestamp.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\type.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\unknown_field_set.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\field_comparator.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\field_mask_util.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\datapiece.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\default_value_objectwriter.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\field_mask_utility.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\json_escaping.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\json_objectwriter.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\json_stream_parser.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\object_writer.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\proto_writer.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\protostream_objectsource.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\protostream_objectwriter.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\type_info.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\type_info_test_helper.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\internal\utility.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\json_util.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\message_differencer.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\time_util.cc" />
+    <ClCompile Include="..\src\google\protobuf\util\type_resolver_util.cc" />
+    <ClCompile Include="..\src\google\protobuf\wire_format.cc" />
+    <ClCompile Include="..\src\google\protobuf\wrappers.pb.cc" />
+  </ItemGroup>
+  <ItemGroup>
+    <ClInclude Include="..\src\google\protobuf\stubs\common.h" />
+    <ClInclude Include="..\src\google\protobuf\stubs\logging.h" />
+  </ItemGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+</Project>
\ No newline at end of file
diff --git a/vsprojects/libprotoc.vcxproj b/vsprojects/libprotoc.vcxproj
new file
index 0000000..ee5aeee
--- /dev/null
+++ b/vsprojects/libprotoc.vcxproj
@@ -0,0 +1,134 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Label="ProjectConfigurations">
+    <ProjectConfiguration Include="Debug|Win32">
+      <Configuration>Debug</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|Win32">
+      <Configuration>Release</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Debug|x64">
+      <Configuration>Debug</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|x64">
+      <Configuration>Release</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>{2B4F3BB4-ADDE-4CB7-B832-C964DE4E0087}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <PropertyGroup>
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
+  <ImportGroup Label="PropertySheets">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+    <Import Project="..\..\..\vsprojects\protobuf.props" />
+  </ImportGroup>
+  <ItemGroup>
+    <ClCompile Include="..\src\google\protobuf\compiler\code_generator.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\command_line_interface.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_enum.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_enum_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_extension.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_file.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_generator.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_helpers.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_map_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_message.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_message_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_primitive_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_service.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\cpp\cpp_string_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_doc_comment.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_enum.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_enum_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_field_base.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_generator.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_helpers.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_map_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_message.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_message_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_primitive_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_reflection_class.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_repeated_enum_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_repeated_message_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_repeated_primitive_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_source_generator_base.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\csharp\csharp_wrapper_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_context.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_doc_comment.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_enum.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_enum_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_enum_field_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_enum_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_extension.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_extension_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_file.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_generator.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_generator_factory.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_helpers.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_lazy_message_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_lazy_message_field_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_map_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_map_field_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_message.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_message_builder.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_message_builder_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_message_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_message_field_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_message_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_name_resolver.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_primitive_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_primitive_field_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_service.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_shared_code_generator.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_string_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\java\java_string_field_lite.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_enum.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_enum_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_extension.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_file.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_generator.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_helpers.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_map_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_message.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_message_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\javanano\javanano_primitive_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\js\js_generator.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_enum.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_enum_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_extension.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_file.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_generator.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_helpers.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_map_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_message.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_message_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_oneof.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\objectivec\objectivec_primitive_field.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\plugin.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\plugin.pb.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\python\python_generator.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\ruby\ruby_generator.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\subprocess.cc" />
+    <ClCompile Include="..\src\google\protobuf\compiler\zip_writer.cc" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="libprotobuf.vcxproj">
+      <Project>{07abd431-1515-4140-932e-e8c150ff7d1a}</Project>
+    </ProjectReference>
+  </ItemGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+</Project>
\ No newline at end of file
diff --git a/vsprojects/protobuf.sln b/vsprojects/protobuf.sln
new file
index 0000000..27d6b04
--- /dev/null
+++ b/vsprojects/protobuf.sln
@@ -0,0 +1,48 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 14
+VisualStudioVersion = 14.0.25123.0
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "libprotobuf", "libprotobuf.vcxproj", "{07ABD431-1515-4140-932E-E8C150FF7D1A}"
+EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "protoc", "protoc.vcxproj", "{DB653289-E887-4B13-AC08-EE3EAC1F34C9}"
+EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "libprotoc", "libprotoc.vcxproj", "{2B4F3BB4-ADDE-4CB7-B832-C964DE4E0087}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Win32 = Debug|Win32
+		Debug|x64 = Debug|x64
+		Release|Win32 = Release|Win32
+		Release|x64 = Release|x64
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{07ABD431-1515-4140-932E-E8C150FF7D1A}.Debug|Win32.ActiveCfg = Debug|Win32
+		{07ABD431-1515-4140-932E-E8C150FF7D1A}.Debug|Win32.Build.0 = Debug|Win32
+		{07ABD431-1515-4140-932E-E8C150FF7D1A}.Debug|x64.ActiveCfg = Debug|x64
+		{07ABD431-1515-4140-932E-E8C150FF7D1A}.Debug|x64.Build.0 = Debug|x64
+		{07ABD431-1515-4140-932E-E8C150FF7D1A}.Release|Win32.ActiveCfg = Release|Win32
+		{07ABD431-1515-4140-932E-E8C150FF7D1A}.Release|Win32.Build.0 = Release|Win32
+		{07ABD431-1515-4140-932E-E8C150FF7D1A}.Release|x64.ActiveCfg = Release|x64
+		{07ABD431-1515-4140-932E-E8C150FF7D1A}.Release|x64.Build.0 = Release|x64
+		{DB653289-E887-4B13-AC08-EE3EAC1F34C9}.Debug|Win32.ActiveCfg = Debug|Win32
+		{DB653289-E887-4B13-AC08-EE3EAC1F34C9}.Debug|Win32.Build.0 = Debug|Win32
+		{DB653289-E887-4B13-AC08-EE3EAC1F34C9}.Debug|x64.ActiveCfg = Debug|x64
+		{DB653289-E887-4B13-AC08-EE3EAC1F34C9}.Debug|x64.Build.0 = Debug|x64
+		{DB653289-E887-4B13-AC08-EE3EAC1F34C9}.Release|Win32.ActiveCfg = Release|Win32
+		{DB653289-E887-4B13-AC08-EE3EAC1F34C9}.Release|Win32.Build.0 = Release|Win32
+		{DB653289-E887-4B13-AC08-EE3EAC1F34C9}.Release|x64.ActiveCfg = Release|x64
+		{DB653289-E887-4B13-AC08-EE3EAC1F34C9}.Release|x64.Build.0 = Release|x64
+		{2B4F3BB4-ADDE-4CB7-B832-C964DE4E0087}.Debug|Win32.ActiveCfg = Debug|Win32
+		{2B4F3BB4-ADDE-4CB7-B832-C964DE4E0087}.Debug|Win32.Build.0 = Debug|Win32
+		{2B4F3BB4-ADDE-4CB7-B832-C964DE4E0087}.Debug|x64.ActiveCfg = Debug|x64
+		{2B4F3BB4-ADDE-4CB7-B832-C964DE4E0087}.Debug|x64.Build.0 = Debug|x64
+		{2B4F3BB4-ADDE-4CB7-B832-C964DE4E0087}.Release|Win32.ActiveCfg = Release|Win32
+		{2B4F3BB4-ADDE-4CB7-B832-C964DE4E0087}.Release|Win32.Build.0 = Release|Win32
+		{2B4F3BB4-ADDE-4CB7-B832-C964DE4E0087}.Release|x64.ActiveCfg = Release|x64
+		{2B4F3BB4-ADDE-4CB7-B832-C964DE4E0087}.Release|x64.Build.0 = Release|x64
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+EndGlobal
diff --git a/vsprojects/protoc.vcxproj b/vsprojects/protoc.vcxproj
new file
index 0000000..5b0afa9
--- /dev/null
+++ b/vsprojects/protoc.vcxproj
@@ -0,0 +1,47 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Label="ProjectConfigurations">
+    <ProjectConfiguration Include="Debug|Win32">
+      <Configuration>Debug</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|Win32">
+      <Configuration>Release</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Debug|x64">
+      <Configuration>Debug</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|x64">
+      <Configuration>Release</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>{DB653289-E887-4B13-AC08-EE3EAC1F34C9}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <PropertyGroup>
+    <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
+  <ImportGroup Label="PropertySheets">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+    <Import Project="..\..\..\vsprojects\protobuf.props" />
+  </ImportGroup>
+  <ItemGroup>
+    <ClCompile Include="..\src\google\protobuf\compiler\main.cc" />
+  </ItemGroup>
+  <ItemGroup>
+    <ClInclude Include="..\src\google\protobuf\compiler\command_line_interface.h" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="libprotoc.vcxproj">
+      <Project>{2b4f3bb4-adde-4cb7-b832-c964de4e0087}</Project>
+    </ProjectReference>
+  </ItemGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+</Project>
\ No newline at end of file
