diff --git a/inc/Windows/pyconfig.h b/inc/Windows/pyconfig.h
index 1cfc59b..d4861cb
--- a/inc/Windows/pyconfig.h
+++ b/inc/Windows/pyconfig.h
@@ -1,6 +1,11 @@
 #ifndef Py_CONFIG_H
 #define Py_CONFIG_H
 
+#ifdef _MSC_VER
+#pragma push_macro("_DEBUG")
+#undef _DEBUG
+#endif
+
 /* pyconfig.h.  NOT Generated automatically by configure.
 
 This is a manually maintained version used for the Watcom,
@@ -756,4 +761,8 @@ Py_NO_ENABLE_SHARED to find out.  Also support MS_NO_COREDLL for b/w compat */
    least significant byte first */
 #define DOUBLE_IS_LITTLE_ENDIAN_IEEE754 1
 
+#ifdef _MSC_VER
+#pragma pop_macro("_DEBUG")
+#endif
+
 #endif /* !Py_CONFIG_H */
